pkg/ide/jupyter.go

```diff
@@ -0,0 +1,274 @@
+			return installPip(hostname)
+		}
+
+		var confirmInstall bool
+		form := huh.NewForm(
+			huh.NewGroup(
+				huh.NewConfirm().
+					Title("pip is not installed. Would you like to install it?").
+					Description("pip is required to install Jupyter Notebook.").
+					Value(&confirmInstall),
+			),
+		).WithTheme(views.GetCustomTheme())
+
+		err := form.Run()
+		if err != nil {
+			return fmt.Errorf("error prompting for pip installation: %w", err)
+		}
+
+		if confirmInstall {
+			return installPip(hostname)
+		} else {
+			return fmt.Errorf("pip is required but not installed")
+		}
+	}
+	views.RenderInfoMessageBold("pip is installed.")
+	return nil
+}
+
+func installPip(hostname string) error {
+	// Check if we're on a Debian-based system
+	if err := runRemoteCommand(hostname, "command -v apt-get"); err == nil {
+		// We're on a Debian-based system, use apt to install pip and python3-venv
+		views.RenderInfoMessageBold("Installing pip and python3-venv using apt...")
+		if err := runRemoteCommand(hostname, "sudo apt-get update && sudo apt-get install -y python3-pip python3-venv"); err != nil {
+			return fmt.Errorf("failed to install pip and python3-venv using apt: %w", err)
+		}
+	} else {
+		// If not Debian-based, fall back to the get-pip.py method
+		views.RenderInfoMessageBold("Installing pip using get-pip.py...")
+		installCmd := "curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 get-pip.py --user && rm get-pip.py"
+		if err := runRemoteCommand(hostname, installCmd); err != nil {
+			return fmt.Errorf("failed to install pip: %w", err)
+		}
+	}
+	return nil
+}
+
+// detectPackageManager detects the package manager on the remote host.
+func detectPackageManager(hostname string) (string, error) {
+	views.RenderInfoMessageBold("Detecting package manager...")
+	commands := map[string]string{
+		"apt-get": "dpkg -s apt >/dev/null 2>&1 && echo apt-get",
+		"yum":     "rpm -q yum >/dev/null 2>&1 && echo yum",
+		"brew":    "brew --version >/dev/null 2>&1 && echo brew",
+	}
+	for manager, cmd := range commands {
+		if err := runRemoteCommand(hostname, cmd); err == nil {
+			return manager, nil
+		}
+	}
+	return "", fmt.Errorf("no supported package manager found")
+}
+
+// installPythonWithPackageManager installs Python using the detected package manager.
+func installPythonWithPackageManager(hostname, manager string) error {
+	views.RenderInfoMessageBold(fmt.Sprintf("Installing Python3 using %s...", manager))
+	var installCmd string
+	switch manager {
+	case "apt-get":
+		installCmd = "sudo apt-get update && sudo apt-get install -y python3"
+	case "yum":
+		installCmd = "sudo yum install -y python3"
+	case "brew":
+		installCmd = "brew install python3"
+	default:
+		return fmt.Errorf("unsupported package manager: %s", manager)
+	}
+	return runRemoteCommand(hostname, installCmd)
+}
+
+// installJupyter installs Jupyter Notebook in a virtual environment on the remote workspace.
+func installJupyter(hostname string) error {
```

Comment:
![Screenshot 2024-09-09 at 13 59 23](https://github.com/user-attachments/assets/eda70c00-8173-4a58-8e2a-074b1de7b30f)

We should have a check if jupyter is installed. This way, every time you run `code` it "installs it again" and that wastes time and bloats the log.

Reply:
You can rename the function to `ensureJupyterInstalled` so it's consistent with the rest of the functions.